File: GameFramePro/Assets/.lpproj/Apis/Unity/CS/GameFramePro/ResourcesEx/IUnityGameObjectPool.lua
---@class CS.GameFramePro.ResourcesEx.IUnityGameObjectPool
CS.GameFramePro.ResourcesEx.IUnityGameObjectPool = {}

---@property readonly CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.BeforGetAction : CS.System.Action
CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.BeforGetAction = nil

---@property readonly CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.BeforeRecycleAction : CS.System.Action
CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.BeforeRecycleAction = nil

---@property readonly CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.PoolContainer : CS.System.Collections.Generic.Stack
CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.PoolContainer = nil

---@property readonly CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.PrefabTarget : CS.UnityEngine.GameObject
CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.PrefabTarget = nil

---@property readonly CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.PoolManagerTarget : CS.UnityEngine.Transform
CS.GameFramePro.ResourcesEx.IUnityGameObjectPool.PoolManagerTarget = nil

---@param capacity : CS.System.Int32
---@param prefabTarget : CS.UnityEngine.GameObject
---@param beforGetAction : CS.System.Action
---@param beforeRecycleAction : CS.System.Action
function CS.GameFramePro.ResourcesEx.IUnityGameObjectPool:InitialedPool(capacity, prefabTarget, beforGetAction, beforeRecycleAction)
end

function CS.GameFramePro.ResourcesEx.IUnityGameObjectPool:ReleasPool()
end

---@return CS.UnityEngine.GameObject
function CS.GameFramePro.ResourcesEx.IUnityGameObjectPool:GetItemFromPool()
end

---@param item : CS.UnityEngine.GameObject
function CS.GameFramePro.ResourcesEx.IUnityGameObjectPool:RecycleItemToPool(item)
end

File: GameFramePro/GameFrame_Client/Assets/Scripts/Core/Module/PoolManager/UnityPool/IUnityGameObjectPool.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;

namespace GameFramePro.ResourcesEx
{
    /// <summary>
    /// 专用于处理Unity MonoBehaviour 对象
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public  interface IUnityGameObjectPool : IUnityObjectPool 
    {
        System.Action<GameObject> BeforeGetAction { get; }
        System.Action<GameObject> BeforeRecycleAction { get; }
        /// <summary>
        /// 用于记录栈中记录元素的hashCode  避免重复缓存
        /// </summary>
        HashSet<int> PollItemRecord { get; }
        Stack<GameObject> PoolContainer { get; }
        GameObject PrefabTarget { get; }
        Transform PoolItemRecycleParent { get; } //池对象管理器自身


        /// <summary>
        /// 初始化
        /// </summary>
        /// <param name="capacity"></param>
        /// <param name="prefabTarget">当对象不足时按照这个创建</param>
        /// <param name="beforGetAction">获取前要进行的操作</param>
        /// <param name="beforeRecycleAction">回收前要做的操作</param>
        void InitialedPool(int capacity, GameObject prefabTarget, System.Action<GameObject> beforGetAction, System.Action<GameObject> beforeRecycleAction);

        /// <summary>
        /// 不需要时候清理自己
        /// </summary>
        void ReleasPool();

        /// <summary>
        /// 获取对象
        /// </summary>
        /// <returns></returns>
        GameObject GetItemFromPool(Transform parent);

        /// <summary>
        /// 回收对象
        /// </summary>
        /// <param name="item"></param>
        void RecycleItemToPool(GameObject item);
    }
}